lib/jobs/redfish-job.js

```diff
@@ -162,33 +163,107 @@ function redfishJobFactory(
     };
     
     /**
-     * Collect Redfish Log Service Entries for resource member
+     * Collect Redfish Log Service Entries for Systems resource member
      * @param redfish client utility
-     * @param resource member to collect entries from
+     * @param system resource path to collect entries from
      */
-    RedfishJob.prototype.collectLogEntries = function(redfish, resource) {
-        return redfish.clientRequest(resource.LogServices['@odata.id'])
+    RedfishJob.prototype.collectSystemLogEntries = function(redfish, resource) {
+        return redfish.clientRequest(resource)
         .then(function(res) {
-            assert.object(res.body, 'LogServices Collection');
-            assert.ok(_.has(res.body, 'Members'));
+            assert.ok(_.has(res.body, 'Members'), 
+                'Has LogServices Members');
             return res.body.Members;
         })
         .map(function(member) {
-            assert.object(member);
+            assert.object(member, 'LogService Member');
             return redfish.clientRequest(member['@odata.id']);
         })
         .map(function(res) {
-            assert.object(res.body, 'LogService Resource');
-            assert.ok(_.has(res.body, 'Entries'));
+            assert.ok(_.has(res.body, 'Entries'),
+                'Has LogService Entries');
             return redfish.clientRequest(res.body.Entries['@odata.id']);
         })
         .map(function(res) {
-            assert.object(res.body, 'Entries Resource');
-            assert.ok(_.has(res.body, 'Items'));
-            return res.body.Items;
+            assert.ok(_.has(res.body, 'Members'), 
+                'Has Entry Members');
+            return res.body.Members;
         })
-        .then(function(data) {
-            data.body = _.flattenDeep(data);
+        .map(function(memberArr) {
+            var promises = [];
+            _.forEach(memberArr, function(member) {
+                promises.push(redfish.clientRequest(member['@odata.id']));
+            });
+            return Promise.all(promises);
+        })
+        .map(function(resArr) {
+            var data = [];
+            _.forEach(resArr, function(res) {
+                data.push(res.body); 
+            });
+            return data;
```

Comment:
just a little suggestion, actually you can simplify this function to a single `_.reduce`